### PR TITLE
Call functions with arbitrary values and retrieve the results

### LINKED ISSUE #29: Synthetically call functions: 8
- Construct simple values
- Build call stack frame
- Break out of trace on function return

### PR DESCRIPTION
**Pull request by rpjohnst**
Closes #29

### CODE REVIEW

File: server/debug/src/call.rs [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+            let (value, float) = write_value(&arg, &arg_type, child, &mut new_context)?;
+
+            let context = new_context.as_raw_mut();
+            if !float {
+                context.R9 = value as winapi::DWORD64;
+            } else {
+                context.FltSave.XmmRegisters[3].Low = value as winapi::ULONGLONG;
+            }
+        }
+
```
Comment (tcourdy) [inline server/debug/src/call.rs:88]: This is pretty heady stuff, so I may have missed it but how are functions with more than 4 arguments handled?

Comment (tcourdy) [reply server/debug/src/call.rs:88]: Oh, will they come through on the stack?

Comment (rpjohnst) [reply server/debug/src/call.rs:88]: Yes- in the Win64 calling convention, the first four arguments are passed in registers RCX, RDX, R8, and R9. Further arguments are passed on the stack.

File: server/debug/examples/call.rs [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+                    debug_continue = true;
+                } else if !first_chance {
+                } else if code == winapi::EXCEPTION_BREAKPOINT {
+                    let thread = threads[&event.thread_id];
+                    let mut context = debug::get_thread_context(thread, winapi::CONTEXT_FULL)
+                        .expect("failed to get context");
+                    context.set_instruction_pointer(address);
+
+                    if let Some((start, start_breakpoint)) = start.take() {
+                        assert_eq!(start, address);
```
Comment (tcourdy) [inline server/debug/examples/call.rs:70]: Does start ever need to be set to none?   I guess I'm confused on when we would hit the else if on line 100 if start is always set in create process.  Thanks.

Comment (rpjohnst) [reply server/debug/examples/call.rs:70]: `.take()` moves the contents out of an `Option` and replaces it with `None`.

**Review by tcourdy — APPROVED** [review]
Looks good.  I think I'm following what is happening.

**Pull request merged**